File: Java/ConsecutiveCharacters.java
class Solution {
    public int maxPower(String s) {
        if (s.length() < 2) return s.length();
        int result = 1;
        int temp = 1;
        for(int i = 0;i<s.length()-1;i++){
            if(s.charAt(i)==s.charAt(i+1)){
                temp++;
            }else{
                result = Math.max(result,temp);
                temp = 1;
            }
        }
        return Math.max(result,temp);
    }
}

File: C#/ConsecutiveCharacters.cs
public class Solution {
    public int MaxPower(string s) {
        if (s.Equals(string.Empty)) return 0;
        if (s.Length == 1) return 1;
        
        char currChar = s[0];
        int maxSubstrLen = 0;
        int cnt = 0;

        for (int i = 0; i < s.Length; i++)
        {
            if (s[i] != currChar)
            {
                currChar = s[i];
                maxSubstrLen = Math.Max(maxSubstrLen, cnt);
                cnt = 0;
            }
            cnt++;
        }
        
        maxSubstrLen = Math.Max(maxSubstrLen, cnt);
        
        return maxSubstrLen;
    }
}